dds/DCPS/TopicDetails.h

```diff
@@ -132,7 +204,18 @@ namespace OpenDDS {
         // Initialize.
         RemoteTopic& remote = remote_topic_iter->second;
         remote.data_type_name_ = type_name;
-        remote.inconsistent_ = topic_callbacks_ && local_data_type_name_ != remote.data_type_name_;
+        remote.type_id_= type_id;
+        if (remote.type_id_.kind() == XTypes::TK_NONE) {
+          remote.inconsistent_ = topic_callbacks_ && local_data_type_name_ != remote.data_type_name_;
+        } else { //deferring remaining checks until can check type_ids for both
+            if (DCPS::DCPS_debug_level) {
```

Comment:
```suggestion
          if (DCPS::DCPS_debug_level) {
```

And the rest of the block

Reply:
You applied this, but the rest of the block needs to be done as well.